File: dqops/src/main/java/com/dqops/core/jobqueue/DqoJobType.java
/*
 * Copyright © 2021-Present DQOps, Documati sp. z o.o. (support@dqops.com)
 *
 * This file is licensed under the Business Source License 1.1,
 * which can be found in the root directory of this repository.
 *
 * Change Date: This file will be licensed under the Apache License, Version 2.0,
 * four (4) years from its last modification date.
 */
package com.dqops.core.jobqueue;

import com.fasterxml.jackson.annotation.JsonProperty;

/**
 * Job type that identifies a job by type.
 */
public enum DqoJobType {
    run_checks,
    run_checks_on_table,
    collect_statistics,
    collect_scheduled_statistics,
    collect_statistics_on_table,
    collect_error_samples,
    collect_error_samples_on_table,
    queue_thread_shutdown,
    synchronize_folder,
    synchronize_multiple_folders,
    run_scheduled_checks_cron,
    import_schema,
    import_tables,
    auto_import_tables,
    delete_stored_data,
    repair_stored_data,
}

File: distribution/python/dqops/client/models/dqo_job_type.py
from enum import Enum


class DqoJobType(str, Enum):
    COLLECT_ERROR_SAMPLES = "collect_error_samples"
    COLLECT_ERROR_SAMPLES_ON_TABLE = "collect_error_samples_on_table"
    COLLECT_STATISTICS = "collect_statistics"
    COLLECT_STATISTICS_ON_TABLE = "collect_statistics_on_table"
    DELETE_STORED_DATA = "delete_stored_data"
    IMPORT_SCHEMA = "import_schema"
    IMPORT_TABLES = "import_tables"
    QUEUE_THREAD_SHUTDOWN = "queue_thread_shutdown"
    REPAIR_STORED_DATA = "repair_stored_data"
    RUN_CHECKS = "run_checks"
    RUN_CHECKS_ON_TABLE = "run_checks_on_table"
    RUN_SCHEDULED_CHECKS_CRON = "run_scheduled_checks_cron"
    SYNCHRONIZE_FOLDER = "synchronize_folder"
    SYNCHRONIZE_MULTIPLE_FOLDERS = "synchronize_multiple_folders"

    def __str__(self) -> str:
        return str(self.value)
